src/JMS/Serializer/Serializer.php

```diff
@@ -121,6 +121,49 @@ public function deserialize($data, $type, $format, DeserializationContext $conte
         return $visitorResult;
     }
 
+    public function toArray($data, SerializationContext $context = null)
+    {
+        if (null === $context) {
+            $context = new SerializationContext();
+        }
+
+        $context->initialize(
+            'json',
+            $visitor = $this->serializationVisitors->get('json')->get(),
+            $this->navigator,
+            $this->factory
+        );
+
+        $visitor->setNavigator($this->navigator);
+        $this->navigator->accept($visitor->prepare($data), null, $context);
+
+        return $visitor->getRoot();
```

Comment:
We need some special handling for the ``ArrayObject`` here that the JSON visitor returns sometimes. Could you add a test for that, too?